programs/margin-account/src/lib.rs

```diff
@@ -1,27 +1,89 @@
 #![feature(proc_macro_hygiene)]
 
 use anchor_lang::prelude::*;
+use anchor_spl::token::{self, TokenAccount};
 
 #[program]
 pub mod margin_account {
     use super::*;
 
-    pub fn initialize(ctx: Context<Initialize>, data: u64) -> ProgramResult {
-        let borrower_account = &mut ctx.accounts.borrower_account;
-        borrower_account.data = data;
+    pub fn initialize(ctx: Context<Initialize>, trader: Pubkey) -> ProgramResult {
+        let margin_account = &mut ctx.accounts.margin_account;
+        margin_account.trader = trader;
+        Ok(())
+    }
+    pub fn deposit(_ctx: Context<Deposit>) -> ProgramResult {
+        // TODO
+        Ok(())
+    }
+    pub fn withdraw(_ctx: Context<Withdraw>) -> ProgramResult {
+        // TODO
+        Ok(())
+    }
+    pub fn trade(_ctx: Context<Trade>) -> ProgramResult {
+        // TODO
+        Ok(())
+    }
+    pub fn liquidate(_ctx: Context<Liquidate>) -> ProgramResult {
+        // TODO
         Ok(())
     }
 }
 
+/// Initializes new margin account.
 #[derive(Accounts)]
 pub struct Initialize<'info> {
     #[account(init)]
-    pub borrower_account: ProgramAccount<'info, BorrowerAccount>,
-    pub rent: Sysvar<'info, Rent>,
+    margin_account: ProgramAccount<'info, MarginAccount>,
+    rent: Sysvar<'info, Rent>,
```

Comment:
I think we can remove this. 

I forgot you create the rent exempt account client side and pass that in as the margin account.

Reply:
But will the chain accept a rent exempt account? Why would one add if you could skip?